### DISCUSSION TITLE
Is it something like cloud9, what's the differences if so?

### DISCUSSION THREAD

**Discussion by conphi** [coder/code-server#674] [General] [answered]

**Comment by pstjvn** [comment]
I will try to answer as best as I can based of my limited knowledge.

I have been using cloud9 for many years up till this spring when ssh work-spaces were shut down in favor of amazon-derived product supposedly providing something "better".

This was my set up:
* headless server (running linux) 
* cloud9 account (notice how the setup depends on having an account with a third party)
* ssh key shared with cloud9 (so it can ssh into my linux box in order to install and run utilities like tmux for example)

This is what I have now with coder:
* headless server (running linux)

and that's it

Coder does not depend on third party account, does not require you to store you ssh keys (and thus giving access) to third parties and cannot be shut down whenever someone buys the third party.

On top of that and because you control the whole setup you can further strengthen the security of the installation (i.e. restricting outgoing connections from the software itself if you are interested in that kind of setup).

I have found VSCode (and thus coder) to be much much better editor/IDE compared to cloud9 which stopped development to cater to modern web devs long time ago, but was still preferable to Vim in many cases. One other great thing I found is that in Chrome it acts as app indeed (so ctrl+w does not close it but closes the tab etc). 

All in all I find it to be much better choice in the cases where you do not want to/can not run local IDE.

**Comment by sr229** [comment, marked as answer]
@conphi There are major differences between C9 and VSCode/Coder.

### Cloud9

Cloud9 used to cater modern web app development through a HTML5 Web IDE. It wasn't full featured but it gets the job done. Cloud9 stopped becoming its own entity when Amazon bought Cloud9's corporate entity, and it became a AWS-only application. While you can run it by yourself, you can't use it commercially nor for Enterprises without specifically buying a license for the IDE.


### Visual Studio Code / Coder

VSCode was a initiative by Microsoft to build a cross-platform editor. It featured the iconic IntelliSense, and has outmatched Atom, another competitor, on the developer adoption. It is more extensible than Cloud9, however the underlying problem with VSCode is the extension host is not really optimized for container usage, nor the entire VSCode codebase. However, it seems to work fine for a 2GB container contrary to Cloud9's relatively small footprint (125MB on use, 65MB idle).

**Comment by sr229** [comment]
CC @Multishifties or @coadler for closure for inactivity.

**Comment by MichaelDesantis** [comment]
@sr229 Just a heads up, I am no longer part of the Coder team and no longer have access to close issues on this repository anymore. 🙁

@coadler Can still close this though. Isn't that right Colin? 🙃
